code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -185,6 +223,74 @@ class ManualCarouselNotificationBuilderTest {
         )
     }
 
+    @Test
+    fun `test to verify notification with action name MANUAL_CAROUSEL_LEFT_CLICKED is having correct center image index`() {
+        every { cacheImages(any()) } answers { 5 }
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        val mockBundle = MockCarousalTemplateDataProvider.getMockedBundleWithManualCarouselData()
+        val data = IntentData(mockBundle, PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED)
+        val mcPushTemplate = CarouselPushTemplate(data) as ManualCarouselPushTemplate
+        val centerIndex = mcPushTemplate.centerImageIndex
+        ManualCarouselNotificationBuilder.construct(
+            context,
+            mcPushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertTrue(mcPushTemplate.centerImageIndex == (centerIndex - 1 + 5) % 5)
```

Comment:
we should verify this with assertEquals(expected value, actual value) so the test failure reason is more descriptive if the test does fail.

Reply:
thinking about the center index calculation further... do we need to calculate the center index with `(centerIndex - 1 + 5) % 5`? we know that we have 5 items in the carousel and for manual carousel, the default center index is 0. if a left click is done we know that the center index will now be 4. I think we can just do a verification of `assertEquals(4, centerIndex)`

Reply:
Replaced the assertTrue to assertEquals 
i've taken the formula to calculate the centre index from the codebase itself, and i'm using 5 because i've created only 5 items, as you can see i'm making the test return only 5 cached images 
` every { cacheImages(any()) } answers { 5 }`